core/src/main/java/org/apache/iceberg/rest/responses/TableScanResponseParser.java

```diff
@@ -106,8 +106,12 @@ public static void serializeScanTasks(
           }
         }
 
-        PartitionSpec partitionSpec = specsById.get(fileScanTask.file().specId());
-        RESTFileScanTaskParser.toJson(fileScanTask, deleteFileReferences, partitionSpec, gen);
+        PartitionSpec spec = specsById.get(fileScanTask.file().specId());
```

Comment:
Should there also be validation on the map itself `specsById` to ensure this is not null otherwise this will throw npe?

Reply:
Yeah I think it's good to add validation for that, but not here imo, I think I'd add it a higher level for each response so that it's clear in the failure that a specific response requires the specs by ID to be set when serializing